src/TreeView/TreeView.js

```diff
@@ -0,0 +1,471 @@
+};
+
+TreeRow.propDescriptions = {
+    isExpanded: '_INTERNAL USE ONLY._',
+    isParent: '_INTERNAL USE ONLY._',
+    rowId: '_INTERNAL USE ONLY._',
+    onExpandClick: '_INTERNAL USE ONLY._'
+};
+
+export class TreeItem extends Component {
+    constructor(props) {
+        super(props);
+
+        const { onExpandClick } = this.props;
+
+        // Generate unique id for row to manage expand/collapse state in parent
+        this.rowId = shortid.generate();
+
+        // Initialize row in parent TreeView state
+        onExpandClick(this.rowId);
+    }
+
+    render() {
+        const {
+            children,
+            expandData,
+            level,
+            onExpandClick,
+            isExpanded: isExpandedProp,
+            ...rest
+        } = this.props;
+        const isExpanded = isExpandedProp || !!expandData[this.rowId];
+
+        // Render child TreeBranch with correct props
+        const childBranch = React.Children.map(children, (child) => {
+            const isTreeBranch = child.type && child.type.name === 'TreeBranch';
+
+            return isTreeBranch ?
+                React.cloneElement(child, {
+                    expandData,
+                    onExpandClick,
+                    isExpanded,
+                    // Increment child branch level
+                    level: level + 1
+                }) :
+                null;
+        });
+
+        // Render child TreeRow with correct props
+        const childRow = React.Children.map(children, (child) => {
+            const isTreeRow = child.type && child.type.name === 'TreeRow';
+
+            return isTreeRow ?
+                React.cloneElement(child, {
+                    isExpanded,
+                    onExpandClick: () => onExpandClick(this.rowId),
+                    isParent: !!childBranch[0],
+                    rowId: this.rowId
+                }) :
+                null;
+        });
+
+        return (
+            <li
+                {...rest}
+                aria-expanded={isExpanded}
+                className='fd-tree__item'
+                id={this.rowId}
+                role='treeitem'>
+                {childRow}
+                {childBranch}
+            </li>
+        );
+    }
+}
+
+TreeItem.displayName = 'TreeItem';
+
+TreeItem.propTypes = {
+    children: PropTypes.node,
+    expandData: PropTypes.object,
+    isExpanded: PropTypes.bool,
+    level: PropTypes.number,
+    onExpandClick: PropTypes.func
+};
+
+TreeItem.defaultProps = {
+    expandData: {},
+    level: 0,
+    onExpandClick: () => {}
+};
+
+TreeItem.propDescriptions = {
+    expandData: '_INTERNAL USE ONLY._',
+    isExpanded: 'Set to *true* for expanded tree item. This variable is handled internally, but can be overridden by the consumer through this prop',
+    level: '_INTERNAL USE ONLY._',
+    onExpandClick: '_INTERNAL USE ONLY._'
+};
+
+export class TreeBranch extends Component {
+    render() {
+        return <Tree {...this.props} />;
```

Comment:
I'm wondering if `Tree` should call `TreeBranch` instead and then move all the conditional class logic and children map to `TreeBranch`.  Or, maybe an internal-only component is created (_e.g._ `_Tree`) that both `Tree` and `TreeBranch` call passing the necessary props?

The reason I say this is because `Tree` is level 0 (by definition) and won't ever be anything other than that.  In fact, `level` shouldn't even be a prop on `Tree` and it should probably be a required prop on `TreeBranch` and _not_ have a default value.

Reply:
Refactored to use an internal `_Tree` component in 74cff8f. 
I couldn't mark the `level` prop as required because im passing that prop through the parent component by iterating through `React.Children` and the required props are validated at creation time of the component so I was getting warnings. I read about it here: https://github.com/facebook/react/issues/4494

Reply:
Yeah, that makes sense.